pg_search/src/query/mod.rs

```diff
@@ -303,7 +303,7 @@ impl SearchQueryInput {
                 let term = Term::from_field_text(field, &value);
                 let distance = distance.unwrap_or(2);
                 let transposition_cost_one = transposition_cost_one.unwrap_or(true);
-                if prefix.unwrap_or(false) {
+                if prefix.unwrap_or(true) {
```

Comment:
Similarly here, it would seem to me that we want `prefix.unwrap_or(false)`, but instead switch the branch bodies below.

My read on the problem here is not that the interface default is wrong, but we have the branches flipped in this `if` block.